x-pack/platform/plugins/shared/streams_app/test/scout/ui/tests/data_management/data_routing/create_routing_rules.spec.ts

```diff
@@ -28,7 +25,7 @@ test.describe('Stream data routing - creating routing rules', { tag: ['@ess', '@
     // Verify we're in the creating new rule state
     await expect(page.getByTestId('streamsAppRoutingStreamEntryNameField')).toBeVisible();
     await expect(page.getByText('Stream name')).toBeVisible();
-    await expect(page.getByText('logs.')).toBeVisible();
+    await expect(page.testSubj.locator('streamNamePrefix')).toHaveText('logs.');
```

Comment:
searching for a text in the whole web page should mot be trusted. Checking specific element

Reply:
Currently we have "77 occurrences in 28 files" with `page.getByText`, so it might require a lot of work to update it.
I share your concern, but let's discuss it to understand how it may impact Teams and if there is always a better alternative.